File: java_solutions/118.pascals-triangle.java
/*
 * @lc app=leetcode id=118 lang=java
 *
 * [118] Pascal's Triangle
 */

class Solution {
    public List<List<Integer>> generate(int n) {
        return Stream.iterate( List.of(1), prev -> IntStream.range(0, prev.size() + 1) .mapToObj(i -> (i == 0 || i == prev.size()) ? 1 : prev.get(i - 1) + prev.get(i)) .toList() ) .limit(n) .toList();
    }
}

File: kotlin_solutions/118.pascals-triangle.kt
/*
 * @lc app=leetcode id=118 lang=kotlin
 *
 * [118] Pascal's Triangle
 */

class Solution { fun generate(numRows: Int): List<List<Int>> { return generateSequence(listOf(1)) { current -> List(current.size + 1) { current.getOrElse(it - 1) { 0 } + current.getOrElse(it) { 0 } } }.take(numRows).toList() } }

